packages/cactus-plugin-keychain-aws-sm/src/main/typescript/plugin-keychain-aws-sm-remote-adapter.ts

```diff
@@ -0,0 +1,154 @@
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(opts, `${fnTag} arg options`);
+    Checks.truthy(opts.keychainId, `${fnTag} arg options.keychainId`);
+    Checks.truthy(opts.instanceId, `${fnTag} options.instanceId`);
+    Checks.nonBlankString(opts.keychainId, `${fnTag} options.keychainId`);
+    Checks.truthy(opts.backend, `${fnTag} options.backend`);
+    Checks.truthy(
+      opts.backend instanceof DefaultApi,
+      `${fnTag} opts.backend instanceof DefaultApi`,
+    );
+
+    const level = this.opts.logLevel || "INFO";
+    const label = this.className;
+    this.log = LoggerProvider.getOrCreate({ level, label });
+
+    this.instanceId = this.opts.instanceId;
+    this.backend = opts.backend;
+    this.keychainId = opts.keychainId;
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  /**
+   * Dummy implementation that wires no web services on the host API server
+   * because there is no need. All the functionality is implemented somewhere
+   * else on a host that's accessible through the network to this object
+   * (because this class is a remote adapter not an actual plugin impl.).
+   *
+   * @param _expressApp
+   */
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    return [];
+  }
+
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  public registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    return this.getOrCreateWebServices();
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  public async shutdown(): Promise<void> {
+    return;
+  }
+
+  public rotateEncryptionKeys(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getEncryptionAlgorithm(): string {
+    throw new Error("Method not implemented.");
+  }
+
+  public getKeychainId(): string {
+    return this.keychainId;
+  }
+
+  public async has(key: string): Promise<boolean> {
+    try {
+      await this.backend.getKeychainEntry({ key });
+      return true;
+    } catch (ex) {
+      // FIXME check for errors being thrown due to something other than
+      // the key not being present...
+      return false;
+    }
+  }
+
+  public async get<T>(key: string): Promise<T> {
+    const { data } = await this.backend.getKeychainEntry({ key });
+    // FIXME what to do here? Does it make any sense to have the get() method
+    // of the keychain be generically parameterizable when we know we can only
+    // return a string anyway?
+    return data.value as any;
+  }
+
+  public async set<T>(key: string, value: T): Promise<void> {
+    // FIXME Does it make any sense to have the set() method be generic?
+    await this.backend.setKeychainEntry({ key, value: value as any });
+  }
+
+  public async delete(key: string): Promise<void> {
+    // FIXME Pretty sure aws secret manager can do delete so we don't have to hack it like this
```

Comment:
@jagpreetsinghsasan Can you confirm if this is leftover from the copy or is the AWS SM also not have an API for deleting entries like Vault?

Reply:
I copied it from the vault one because even in vault remote adapter, the other endpoints aren't implemented yet

Reply:
I have updated the newly created issue for adding this endpoint too or maybe update the set-keychain-entry endpoint with some tag to represent deletion.

Reply:
I didn't mean the endpoint implementation, but specifically the delete method's implementation being the way it is where instead of calling a delete method it sets it with an empty string (which is different from calling a delete method on the AWS SDK - if it exists).
So my question is about the underlying libraries that we use, not the plugins themselves.

Reply:
Yes, this will be replaced with deleteKeychainEntry( ) method just like we have to setKeychainEntry( )
![image](https://user-images.githubusercontent.com/56861721/118611407-6dee0100-b7da-11eb-899b-c00a041ac8af.png)

Reply:
But I didnt define the deleteKeychainEntry because its not defined in the openapi of cactus-core-api

Reply:
Hi @petermetz 
I have mentioned this to be fixed in the issue #967, to be fixed with the overall web services endpoints like get and set keychainEntry

Reply:
@jagpreetsinghsasan Could you also open a separate issue to add the delete method to the core-api keychain definitions?

Reply:
I have added that to the issue #967 (third point of the description of the issue) while creating it.